src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorComponentSearchEngine.cs

```diff
@@ -11,6 +9,8 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer
 {
     internal abstract class RazorComponentSearchEngine
     {
-        public abstract Task<DocumentSnapshot> TryLocateComponentAsync(TagHelperDescriptor tagHelper);
+        public abstract Task<DocumentSnapshot?> TryLocateComponentAsync(TagHelperDescriptor tagHelper);
```

Comment:
Ah ya we were bad before too and didn't pass in cancellation tokens. One of these days we should revisit all of our async methods and just add em